pkg/apis/clouddns/v1/zone_genclient.go

```diff
@@ -47,13 +47,12 @@ func (z *Zone) FilterAPIRequestBody(ctx context.Context) (interface{}, error) {
 	if op == types.OperationCreate || op == types.OperationUpdate {
 		zWithZoneName := struct {
 			Zone
-			ZoneName string `json:"zoneName"`
+			// for these operations the Engine expects the name to be in zoneName instead of name
+			// first we had the additional ZoneName attribute, but we can just rename the name attribute
+			// instead.
+			Name string `json:"zoneName"`
```

Comment:
I checked the resulting JSON with `json.NewEncoder(os.Stdout).Encode(zWithZoneName)` and it still contains a name field - so I think this re-annotating is not replacing the existing annotation, but extending it.

Reply:
yeah, you are right - this way it adds a field and not replaces it